### ISSUE TITLE
[FR]

### ISSUE THREAD

**Issue by Alec-Stashevsky** [mlflow/mlflow#5383]
Thank you for submitting a feature request. **Before proceeding, please review MLflow's [Issue Policy for feature requests](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md#feature-requests) and the [MLflow Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst)**.

**Please fill in this feature request template to ensure a timely and thorough response.**

## Willingness to contribute
The MLflow Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature (either as an MLflow Plugin or an enhancement to the MLflow code base)?

- [ ] Yes. I can contribute this feature independently.
- [ X] Yes. I would be willing to contribute this feature with guidance from the MLflow community.
- [ ] No. I cannot contribute this feature at this time.

## Proposal Summary

(In a few sentences, provide a clear, high-level description of the feature request)

## Motivation
- What is the use case for this feature?
To be able to download time-series of metrics logged to mlflow as a csv. Whatever data can be visualized on the line plot in the mlflow experiment tracker Metrics pane will able to be download as csv for more detailed analysis or visualization. 
- Why is this use case valuable to support for MLflow users in general?
Metrics time series can be very useful for model selection and diagnosis. By viewing the time-corrected covariation between metrics, it will allow users to gain further insight into the model. Additionally, having a csv allows users flexibility to visualize and analyze these valuable metrics in whichever method they choose.
- Why is this use case valuable to support for your project(s) or organization?
Gaining further insight into the model selection process. Additionally to help understand what metrics may be redundant or misleading.
- Why is it currently difficult to achieve this use case? (please be as specific as possible about why related MLflow features and components are insufficient)
The data is already stored somewhere in mlflow, but it is just not easily accessible to the user.

### What component(s), interfaces, languages, and integrations does this feature affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interfaces
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Languages 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

## Details

(Use this section to include any additional information about the feature. If you have a proposal for how to implement this feature, please include it here. For implementation guidelines, please refer to the [Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#contribution-guidelines).)

Just make whatever data is used to create existing visualization available to the user for download. I think it would be a very small adjustment.

**Comment by dbczumar** [comment]
Hi @Alec-Stashevsky. Thank you for this suggestion! Sounds like a great idea! Would you be able to draft a PR for this? I think we can reuse the existing UI assets and much of the existing logic from the "Download" button on the Experiment Page.

**Comment by Alec-Stashevsky** [comment]
@dbczumar I would like to but I am quite unfamiliar with the library. Could you point me to (1) where that data is stored / called to build the existing metrics plots and (2) the existing logic for the Download button?

**Comment by dbczumar** [comment]
@Alec-Stashevsky Apologies for the delay. Here's the code for the Download button: https://github.com/mlflow/mlflow/blob/ddfcccc631b56cae8083dbb00510d1215a1deded/mlflow/server/js/src/experiment-tracking/components/ExperimentView.js#L698-L703. The associated onClick function, `onDownloadCsv()` (https://github.com/mlflow/mlflow/blob/ddfcccc631b56cae8083dbb00510d1215a1deded/mlflow/server/js/src/experiment-tracking/components/ExperimentView.js#L1215-L1233) has a lot of relevant code for getting started.

**Comment by Alec-Stashevsky** [comment]
@dbczumar thanks for the help! Unfortunately after taking a look around I am unable to contribute to the feature at this time. I have edited the request above.

**Comment by Jingnan-Jia** [comment]
Any update now? I really expect this feature !

**Comment by Alec-Stashevsky** [comment]
@dbczumar any chance on this being implemented anytime soon?

**Comment by dbczumar** [comment]
Hi @Alec-Stashevsky @Jingnan-Jia , I've added the `help wanted` and `good first issue` labels to the issue in hopes of identifying a contributor.

**Comment by subramaniam02** [comment]
I am interested in taking this up. I am newbie to mlflow project and interested in learning about mlflow by working on this issue. Hope there is no time pressure to get this feature out ASAP.

**Comment by dbczumar** [comment]
@subramaniam02 Sounds wonderful! Thank you in advance for your contribution, and let us know if you have any questions! :)

**Comment by subramaniam02** [comment]
@dbczumar  @Alec-Stashevsky  Please help me understand the requirements

- Existing download button provides only the latest value of metrics for each run of the experiment ?
- Support downloading of all metrics (in csv format) for each run of the experiment (individual run) or for all the runs of the experiment?
- There should be new download button to download the metrics seen in the graph like the screenshot ?

![Screen Shot 2022-06-02 at 3 41 51 PM](https://user-images.githubusercontent.com/1637727/171750242-5f7a1877-09dc-44d0-bdc4-796f08e73481.png)

**Comment by Alec-Stashevsky** [comment]
@subramaniam02 yes that is the idea. My intention was to have the download button be experiment-specific. So there would be a download button on the page you display above, and it would download the entire time-series of metrics (recorded at each step) for that experiment. 

That csv would basically contain all the data needed to make the plot mlflow displays. It would contain the data for the x-axis  (Step, Time (Wall), and Time (Relative). It would also contain on the data for the y-axis, i.e., all the metrics recorded for that given experiment.

**Comment by rafaelvp-db** [comment]
Created #6048 for this. @dbczumar would you mind reviewing?

**Comment by Alec-Stashevsky** [comment]
Looks good to me, just wanted to bump @rafaelvp-db's PR. 

Thanks!
